ibis/expr/types/joins.py

```diff
@@ -103,13 +103,14 @@ def prepare_predicates(
     for pred in util.promote_list(predicates):
         if pred is True or pred is False:
             yield ops.Literal(pred, dtype="bool")
-        elif isinstance(pred, ValueExpr):
-            node = pred.op()
-            yield dereference_value(node, deref_left, deref_right)
+        elif isinstance(pred, Value):
+            for node in flatten_predicates(pred.op()):
+                yield dereference_value(node, deref_left, deref_right)
```

Comment:
This change is not essential to the fix itself, but flattening must happen before trying to dereference the sides.